subsys/net/ip/Kconfig.ipv6

```diff
@@ -159,7 +159,7 @@ config NET_IPV6_DAD
 config NET_IPV6_RA_RDNSS
 	bool "Support RA RDNSS option"
 	depends on NET_IPV6_ND
-	select DNS_RESOLVER
+	depends on DNS_RESOLVER
```

Comment:
Nice, I wondered exactly this weirdness in #72834 just few minutes ago. Good that it is now fixed.

Reply:
@rlubos Wow! What an unbelievable coincidence - just found the same issue and tried disabling NET_IPV6_RA_RDNSS, as I don't use DNS_RESOLVER in my app. Your solution is definitely better and more desirable. Thank you!